lib/Dialect/Verif/VerifOps.cpp

```diff
@@ -58,82 +58,99 @@ OpFoldResult HasBeenResetOp::fold(FoldAdaptor adaptor) {
     if (auto enableConst = enable.getDefiningOp<hw::ConstantOp>())
       if (enableConst.getValue().isZero()) {
         rewriter.eraseOp(op);
         return success();
       }
+  return failure();
+}
 
-  // verif.assert %a where %a is ltl.boolean_constant true -> <delete>
+/// Delete operation if property is `ltl.boolean_constant true`.
+template <typename Op>
+static LogicalResult eraseIfPropertyTrue(Op op, PatternRewriter &rewriter) {
   Value property = op.getProperty();
   if (auto boolConst = property.getDefiningOp<ltl::BooleanConstantOp>())
     if (boolConst.getValueAttr().getValue()) {
       rewriter.eraseOp(op);
       return success();
     }
-
   return failure();
 }
 
-} // namespace AssertLikeOp
+/// Convert `op(ltl.clock(prop, clk), en)` to `clocked_op(prop, en, clk)`.
+template <typename TargetOp, typename Op>
+static LogicalResult lowerToClocked(Op op, PatternRewriter &rewriter) {
+  Value property = op.getProperty();
+  auto clockOp = property.getDefiningOp<ltl::ClockOp>();
+  if (!clockOp)
+    return failure();
 
-LogicalResult AssertOp::canonicalize(AssertOp op, PatternRewriter &rewriter) {
-  if (succeeded(AssertLikeOp::canonicalize(op, rewriter)))
-    return success();
+  rewriter.replaceOpWithNewOp<TargetOp>(
+      op, clockOp.getInput(), ltlToVerifClockEdge(clockOp.getEdge()),
+      clockOp.getClock(), op.getEnable(), op.getLabelAttr());
+  return success();
+}
 
-  return CoverLikeOp::canonicalize<ClockedAssertOp>(op, rewriter);
+LogicalResult AssertOp::canonicalize(AssertOp op, PatternRewriter &rewriter) {
+  return success(succeeded(eraseIfEnableFalse(op, rewriter)) ||
+                 succeeded(eraseIfPropertyTrue(op, rewriter)) ||
+                 succeeded(removeEnableTrue(op, rewriter)) ||
+                 succeeded(lowerToClocked<ClockedAssertOp>(op, rewriter)));
 }
 
 LogicalResult AssumeOp::canonicalize(AssumeOp op, PatternRewriter &rewriter) {
-  if (succeeded(AssertLikeOp::canonicalize(op, rewriter)))
-    return success();
-
-  return CoverLikeOp::canonicalize<ClockedAssumeOp>(op, rewriter);
+  return success(succeeded(eraseIfEnableFalse(op, rewriter)) ||
+                 succeeded(eraseIfPropertyTrue(op, rewriter)) ||
+                 succeeded(removeEnableTrue(op, rewriter)) ||
+                 succeeded(lowerToClocked<ClockedAssumeOp>(op, rewriter)));
 }
 
 LogicalResult CoverOp::canonicalize(CoverOp op, PatternRewriter &rewriter) {
-  return CoverLikeOp::canonicalize<ClockedCoverOp>(op, rewriter);
+  return success(succeeded(removeEnableTrue(op, rewriter)) ||
+                 succeeded(lowerToClocked<ClockedCoverOp>(op, rewriter)));
+}
+
+LogicalResult ClockedAssertOp::canonicalize(ClockedAssertOp op,
+                                            PatternRewriter &rewriter) {
+  return success(succeeded(removeEnableTrue(op, rewriter)) ||
+                 succeeded(eraseIfEnableFalse(op, rewriter)) ||
+                 succeeded(eraseIfPropertyTrue(op, rewriter)));
+}
```

Comment:
nit: Can we register these patterns like https://github.com/llvm/circt/blob/bf970671340c099008188c24f3ba8b2aada6c7e7/lib/Dialect/FIRRTL/FIRRTLFolds.cpp#L2601-L2605 within getCanonicalizationPatterns instead of chaining these patterns in canonicalize method?